src/libs/NextStepUtils.ts

```diff
@@ -270,6 +270,62 @@ function buildNextStep(report: OnyxEntry<Report>, predictedNextStatus: ValueOf<t
 
         // Generates an optimistic nextStep once a report has been approved
         case CONST.REPORT.STATUS_NUM.APPROVED:
+            if (PolicyUtils.hasAdvancedApprovalEnabled(policy)) {
+                const approvalChain = PolicyUtils.getApprovalChain(policy, currentUserAccountID, report?.total ?? 0);
+                if (approvalChain[0] === currentUserEmail) {
```

Comment:
FYI I believeeeee you can approve your own report as more than just the 0th approver

Example:

Employee A submitsTo Employee B
Employee B forwardsTo Employee A

Reply:
https://expensify.slack.com/archives/C06ML6X0W9L/p1722560950570639?thread_ts=1722319732.404769&cid=C06ML6X0W9L